### PR TITLE
Add Package occt

### PR DESCRIPTION
**Pull request by craffael**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/create/cmake.html)
  step by step carefully. **Yes**

* My change will resolve this "New package" request: https://github.com/cpp-pm/hunter/issues/222

### CODE REVIEW

File: cmake/projects/occt/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+include(hunter_pick_scheme)
+include(hunter_cmake_args)
+
+hunter_add_version(
+    PACKAGE_NAME
+    occt
+    VERSION
+    7.4.0-p1
+    URL
+    "https://github.com/cpp-pm/OCCT/archive/hunter-7.4.0-p1.tar.gz"
```
Comment (rbsheth) [inline cmake/projects/occt/hunter.cmake:18]: You aren't supposed to create random branches for packages. The correct form is:
hunter-<version>-<patch version>
I released `v7.4.0-p0` from the branch `hunter-7.4.0`, which I merged your incorrectly named `hunter-7.4.0-p1` named branch into. I deleted the incorrect branch.

Please use this release here: https://github.com/cpp-pm/OCCT/releases/tag/v7.4.0-p0

File: docs/packages/pkg/occt.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+.. _pkg.occt:
+
+occt
+====
+OpenCascade Community Technology
+
+-  `Official <https://dev.opencascade.org/>`__
+-  `Hunterized <https://github.com/cpp-pm/occt>`__
+-  `Example <https://github.com/cpp-pm/hunter/blob/master/examples/occt/CMakeLists.txt>`__
+-  Added by `craffael <https://github.com/craffael>`__ (`pr-N <https://github.com/cpp-pm/hunter/pull/N>`__)
```
Comment (rbsheth) [inline docs/packages/pkg/occt.rst:17]: Please update PR number here

File: examples/occt/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+find_package(OpenCASCADE CONFIG REQUIRED)
+
+add_executable(boo boo.cpp)
+target_link_libraries(boo occt::TKFillet  occt::TKOffset occt::TKBool  occt::TKBO occt::TKShHealing  occt::TKPrim occt::TKTopAlgo occt::TKBRep occt::TKGeomAlgo occt::TKGeomBase occt::TKG3d occt::TKG2d occt::TKMath occt::TKernel)
+if(APPLE OR (UNIX AND NOT ANDROID))
+  target_link_libraries(boo pthread)
+elseif(WIN32)
+  target_link_libraries(boo ws2_32)
+endif()
+# DOCUMENTATION_END }
```
Comment (rbsheth) [inline examples/occt/CMakeLists.txt:23]: Nits: Add a single newline at the end of files

**Review by rbsheth — CHANGES_REQUESTED** [review]
Thanks for the PR! This must've been a lot of work 😅 

Couple issues to resolve.

File: docs/packages/pkg/occt.rst (hunk 2 above)
Comment (craffael) [reply docs/packages/pkg/occt.rst:17]: Done :)

File: examples/occt/CMakeLists.txt (hunk 3 above)
Comment (craffael) [reply examples/occt/CMakeLists.txt:23]: done

**Comment by craffael** [comment]
Thanks for the review. It was indeed a lot of work, mostly because OCCT doesn't really use a modern cmake style + everytime I wanted to test if it compiles it took 1h+.

**Review by rbsheth — APPROVED** [review]

**Comment by rbsheth** [comment]
@craffael Thank you!

**Comment by rahulshethsc** [comment]
Released: https://github.com/cpp-pm/hunter/releases/tag/v0.23.280

**Pull request merged**